packages/block-library/src/post-comments-count/edit.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * WordPress dependencies
+ */
+import { useEntityId } from '@wordpress/core-data';
+import { useState, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+function PostCommentsCountDisplay() {
+	const postId = useEntityId( 'postType', 'post' );
+	const [ commentsCount, setCommentsCount ] = useState();
+	useEffect( () => {
```

Comment:
I've been using `react-fetching-library` recentlly and I believe we can learn from its API and propose a `useApiFetch` hook.

```
const { loading, payload, response, query } = useApiFetch( request );
```

Reply:
What do you think of https://github.com/zeit/swr?

Reply:
It's a very similar idea, I tried both in a project. I preferred the "react-fetching-library" approach because of the way they consider basically requests like "redux actions and mutations like redux action creators". It felt like a very natural API to use but I don't hold strong preferences here, the idea is just to absorb the boilerplate which is always the same when you want to trigger an API + have some caching.

Reply:
Mmm, it seems less flexible. E.g., mutations and queries use a different syntax, and I don't see a first-class way to perform local mutation optimizations.

We need to standardize the API so everything can be entities. This is only here because the required data is in a response header.